### PR TITLE
complete ticket 5 - Added PATCH method to the reviews endpoint

### PR DESCRIPTION
**Pull request by MattArwenLangham**
This pull request completes ticket 5.

This branch adds:

- handles a POST request for the reviews endpoint, updating the views of a specified review_id by a submitted amount.
- error handling for invalid ID, incorrect data type, incorrect property(not inc_view) and incorrect value data type.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 
 const express = require("express");
 const app = express();
 
+app.use(express.json())
+
 app.get("/api/categories", getCategories);
 
 app.get("/api/reviews/:review_id", getReviewById)
+app.post("/api/reviews/:review_id", postReview)
```
Comment (karlaferry) [inline app.js:14]: This is a patch request. You will have a ticket to post an actual review (I think), but for now we are just patching the votes of the review.

File: __tests__/nc-games.test.js [hunk 2]
```diff
@@ -85,4 +85,84 @@ describe("/api/reviews/:review_id", () => {
             }) 
         })
     })
+
+    describe("PATCH", () => {
+        test("Status 201: When a valid review_id is passed, if a positive number is passed, increment it by that much", () => {
+
+            return request(app)
+            .post("/api/reviews/2")
```
Comment (karlaferry) [inline __tests__/nc-games.test.js:93]: Patch request please.

**Review by karlaferry — CHANGES_REQUESTED** [review]
Please change all reference to `post` in this PR. We are only doing patches for now :) Then send a new PR again for approval :)

File: __tests__/nc-games.test.js (hunk 2 above)
Comment (MattArwenLangham) [reply __tests__/nc-games.test.js:93]: That is my 4pm brain! Thanks for bringing that up, totally missed it! Will resubmit it shortly.

**Pull request closed without merging**